Saml/Saml.cs

```diff
@@ -287,31 +290,45 @@ public string GetCompany()
 
 		public List<string> GetGroups()
 		{
-			// this is only valid for azure claims.
-			var node_list = _xmlDoc.SelectNodes("/samlp:Response/saml:Assertion[1]/saml:AttributeStatement/saml:Attribute[@Name='http://schemas.microsoft.com/ws/2008/06/identity/claims/groups']/saml:AttributeValue", _xmlNameSpaceManager);
-			var group_ids = new List<string>();
-			foreach (XmlNode node in node_list) {
-				group_ids.Add(node.InnerText);
+			foreach (string name in CoalesceNamesAndPrefixedNames(new List<string> { "groups" }, GROUPS_MS_NAME_PREFIX)) {
+				var attempt = GetAllCustomAttribute(name);
+				if (attempt.Count() > 0)
+					return attempt.ToList();
 			}
-			return group_ids;
+			return new List<string>();
 		}
 
+		private IEnumerable<string> ApplyNamePrefixToAll(IEnumerable<string> names, string name_prefix) => names.Select(name => name_prefix + name);
+
+		private IEnumerable<string> CoalesceNamesAndPrefixedNames(IEnumerable<string> names, string name_prefix) => names.Union(ApplyNamePrefixToAll(names, name_prefix));
+
 		private string AssertionAttributeValueForWithCoalesce(params string[] names_to_try)
 		{
-			foreach (string name in names_to_try.Union(names_to_try.Select(name => $"http://schemas.xmlsoap.org/ws/2005/05/identity/claims/{name}"))) {
+			foreach (string name in CoalesceNamesAndPrefixedNames(names_to_try, NORMAL_ATTRIBUTE_NAME_PREFIX)) {
 				var attempt = GetCustomAttribute(name);
 				if (attempt != null)
 					return attempt;
 			}
 			return null;
 		}
 
-		public string GetCustomAttribute(string attr)
+		private string AttributeByNameXPath(string name) => $"/samlp:Response/saml:Assertion[1]/saml:AttributeStatement/saml:Attribute[@Name='{name}']/saml:AttributeValue";
+
+		private string GetCustomAttribute(string attr)
```

Comment:
Although I think you are right to make it private, since it is not in the interface, we should probably keep it public since this package is open source (and public on Nuget too) and it might be a breaking change for someone.

Reply:
Good call will change back to public